### DISCUSSION TITLE
Trying to create multiple khepri clusters in the same Elixir application, using custom RaServerConfig.

### DISCUSSION THREAD

**Discussion by sksar** [rabbitmq/khepri#284] [Q&A] [answered]
I was trying to create multiple khepri clusters using RaServerConfig, in an elixir application, but I can't really find any good documentation regarding the same.

This I am doing beacuse I need khepri to start multiple Ra clusters with different names, but in elixir it is not happening by default. So RaServerConfig is what i was trying to do.

```
 :khepri.start(".config", %{
      name: "config",
    })
```

Can anyone guide me ?

**Comment by lukebakken** [comment]
Hello! It would be much easier to assist you if you provided your code via a git repository we could clone. 3 lines of code is insufficient.

**Comment by sksar** [comment]


**Reply by lukebakken** [reply]
Please take the time to distill what you are trying to do into a runnable example. If you expect to get support from us for _FREE_, the least you can do is put some effort into demonstrating your issue.

There is a chance that @the-mikedavis or @kjnilsson can figure this out just by looking at your code.

**Reply by lukebakken** [reply]
This is just after 5 minutes of khepri code browsing. Please note the dialyzer specs to determine the correct types for function arguments:

https://github.com/rabbitmq/khepri/blob/main/src/khepri_cluster.erl#L218-L229
https://github.com/rabbitmq/khepri/blob/main/src/khepri_cluster.erl#L245-L274

```elixir
:khepri.start("/full/path/to/existing/data/dir", :my_khepri_store_1)
```

**Comment by dumbbell** [comment]
To start multiple Kheri stores, you need to call `khepri:start()` for each of them. This function and its parameters are documented. You can use [`khepri:start/2`](https://rabbitmq.github.io/khepri/khepri.html#start-2) to specify both the data directory and the name of the store:

```erlang
khepri:start("/path/to/store1", store1),
khepri:start("/path/to/store2", store2),
```

(I will let you convert that to Elixir)

**Reply by michaelklishin** [reply]
@sksar we do not take problem definitions such as "it does not work" seriously. Please put together a repository with an executable way to reproduce the behavior or take your questions elsewhere.

```
mix new /path/to/khepri-question-284
cd /path/to/khepri-question-284
# put together an example
git add .
git commit -m "Some code to reproduce the behavior from rabbitmq/khepri#284"
# push the repo to a public repo on GitHub
# share the link to said public repo
```

**Comment by sksar** [comment, marked as answer]
#279 will solve the issue.
I have found the issue. Thank you all.
